### PR TITLE
Add CriticalAddonsOnly toleration to all k8s 1.6 kubeadm manifests.

### LINKED ISSUE #705: Missing toleration from 1.6 kubeadm calico manifest
This file:
http://docs.projectcalico.org/master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml

seems to be missing this toleration:
```
      # Mark the pod as a critical add-on for rescheduling.
      - key: CriticalAddonsOnly
        operator: Exists
```

Without it, I was seeing that my cluster was only starting calico-node on the master...

### PR DESCRIPTION
**Pull request by fasaxc**
Fixes #705

### CODE REVIEW

**Comment by tmjd** [comment]
@fasaxc have you tested this manually?

**Comment by fasaxc** [comment]
@tmjd No, @lwr20 said it was needed for one manifest and then I back-ported it.  I think when the annotations were replaced with real tolerations, we must have dropped the `CriticalAddonsOnly` one.

**Comment by djosborne** [comment]
Can you add a comment to the yaml explaining why this is needed? i.e. which clusters get that master taint? Kubeadm?

**Comment by tmjd** [comment]
I brought up a cluster with kubeadm, installed the manifest from @fasaxc's branch from the v2.1/..../kubeadm/1.6/calico.yaml, killed a calico/node pod and a new one was rescheduled successfully.
Test complete unless someone thinks there is more that needs done.

**Comment by fasaxc** [comment]
@tmjd I've now added the comments that we discussed.  Could you give at least one of the manifests another spin just to check I didn't typo?

**Comment by tmjd** [comment]
Still checks out. LGTM

File: master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml [hunk 1]
```diff
@@ -54,12 +54,19 @@ spec:
       labels:
         k8s-app: calico-etcd
       annotations:
+        # Mark this pod as a critical add-on; when enabled, the critical add-on scheduler
+        # reserves resources for critical add-on pods so that they can be rescheduled after
+        # a failure.  This annotation works in tandem with the toleration below.
         scheduler.alpha.kubernetes.io/critical-pod: ''
```
Comment (djosborne) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml]: Is the annotation still needed for 1.6? I would imagine it only needs the actual toleration (and I'd like to start cutting out the cruft that is leftover toleration-by-annotations in our 1.6 manifests)

Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml]: The Kubernetes docs @fasaxc linked in the original issue say both are needed.
https://kubernetes.io/docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods/

Comment (fasaxc) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml]: @djosborne Just to close the loop, there used to be two annotations, this one to mark our pod as a critical add-on and a second one for the old format of toleration.  this one is still needed, but the toleration one has moved to the "real" toleration syntax.

**Pull request merged**